build-windows.rb

```diff
@@ -40,13 +40,13 @@ def get_zynaddsubfx()
     cmd   "git submodule update --init"
     chdir ".."
     cmd   "mkdir -p pkg"
-    cmd   "sh ./z/build-fftw.sh"
+    cmd   "sudo sh ./z/build-fftw.sh"
     #cmd   "sh ./z/build-jack.sh"
-    cmd   "sh ./z/build-liblo.sh"
-    cmd   "sh ./z/build-mxml.sh"
-    cmd   "sh ./z/build-portaudio.sh"
-    cmd   "sh ./z/build-zlib.sh"
-    cmd   "cp ./libwinpthread* ./pkg/bin/"
+    cmd   "sudo sh ./z/build-liblo.sh"
+    cmd   "sudo sh ./z/build-mxml.sh"
+    cmd   "sudo sh ./z/build-portaudio.sh"
+    cmd   "sudo sh ./z/build-zlib.sh"
+    cmd   "sudo cp /usr/x86_64-w64-mingw32/lib/libwinpthread-1.dll ./pkg/bin/"
```

Comment:
None of these scripts require sudo. The only place sudo should be needed is installing the cross compiling system packages through apt. All of the zyn specific dependencies are built and installed to a non-root directory.

Reply:
The `make` and `./configure` of these scripts still require to be ran with `sudo`, or it wil fail due to permission problems. So I either need to add `sudo` before the `./configure` and `make` or run the script with `sudo`.
Note that im using a fresh Ubuntu 16.04 default install. The permission problem is probaly the **main** cause why people fail to build zynfusion.
Back when I managed to compile zyn-fusion, I had to add `sudo` before every `./configure` and `make`. Without these changes the buildscript will fail.

So the sudo is necessary.

Reply:
There might be a problem somewhere, but I don't follow that line of logic. Take mxml as an example. It is downloaded to a non-root owned directory. Then it is untarred in a non-root owned directory. Then the configure scripts are called, which don't use any root permissions and the configure scripts set the install path to a non-root directory. Then the software is built (fully in a non-root directory). Then it installs to the pkg directory which was previously created as a non-root directory.

At no step in that process is root permissions required.

One way that root permissions *would* be required is if the build-windows.rb script was ever run as root. If that happened, then the 'pkg' directory would be owned by the root user causing this process to fail. Note that in my opinion the build-windows.rb script should never be run as root to avoid this sort of problem.

Reply:
For some reason the pkg and all the downloaded dependecies folders required root accest to edit, I dont know why. Deleted and Cloned my fork again, and now the issue is gone. Thank you for your patience.

Reply:
For some reason cmake now fails to find pthread_create, still no idea why.

Reply:
not sure what would cause issues with detecting pthread.

Reply:
The wierd thing is, when I run the cmake command without sudo, if finds all packages but fails due to missing write permissions for the tests, but when i run the command with sudo afterwards, it builds just fine.
When just ran with sudo, it says it cant find any package and the cmake error.txt says finding pthread_create failed.

Reply:
I managed to resolve the permission problem. It seemed that CMakeFiles directory required root access, but that should be fixed now.But still the find pthread_create problem persists. I attached the log, maybe you see something I missed.
[CMakeError.log](https://github.com/zynaddsubfx/zyn-fusion-build/files/1844384/CMakeError.log)

Reply:
Why would the CMakeFiles directory be owned by root unless you've built the package as root in the past?

Reply:
Basicely there the build-zynaddsubfx-#demomode directory wasnt deleted so i ran the cmake command with sudo to check if it works and that made the directory owned by root. I had to delete the folder to make it work again. Also added it to the build-windows.rb to avoid this problem in the future.